client/my-sites/checkout/composite-checkout/components/checkout-next-steps.tsx

```diff
@@ -0,0 +1,129 @@
+import styled from '@emotion/styled';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { ReactChild, useMemo } from 'react';
+import { hasDIFMProduct } from 'calypso/lib/cart-values/cart-items';
+
+interface Props {
+	responseCart: ResponseCart;
+	headerText?: TranslateResult;
+}
+
+interface NextStepsStep {
+	text: ReactChild;
+	icon: JSX.Element;
+}
+
+const CheckoutNextStepsWrapper = styled.div`
+	background: ${ ( props ) => props.theme.colors.surface };
+	border: 1px solid ${ ( props ) => props.theme.colors.borderColorLight };
+	margin: 0;
+	padding: 20px;
+`;
+
+const CheckoutNextStepsHeader = styled.h3`
+	font-size: 14px;
+	font-weight: ${ ( props ) => props.theme.weights.bold };
+	margin-bottom: 16px;
+`;
+
+const CheckoutNextStepsListWrapper = styled.ul`
+	margin: 0;
+	list-style: none;
+	font-size: 14px;
+`;
+
+const CheckoutNextStepsListItem = styled( 'li' )`
+	margin-bottom: 4px;
+	overflow-wrap: break-word;
+	display: flex;
+	align-items: flex-start;
+	margin-bottom: 12px;
+
+	.rtl & {
+		padding-right: 24px;
+		padding-left: 0;
+	}
+`;
+
+const BaseIcon = styled.div`
+	background: ${ ( props ) => props.theme.colors.success };
+	border-radius: 50%;
+	border: 1px solid;
+	border-color: ${ ( props ) => props.theme.colors.success };
+	height: 18px;
+	width: 18px;
+	text-align: center;
+	flex-shrink: 0;
+	margin-right: 8px;
+
+	.gridicon {
+		fill: ${ ( props ) => props.theme.colors.surface };
+	}
+`;
+
+const CompletedStepIcon = () => (
+	<BaseIcon>
+		<Gridicon icon="checkmark" size={ 12 } />
+	</BaseIcon>
+);
+
+const CurrentStepIcon = styled( BaseIcon )`
+	background: ${ ( props ) =>
+		`linear-gradient(0, ${ props.theme.colors.success } 50%, ${ props.theme.colors.surface } 50%)` };
+`;
+
+const NextStepIcon = styled( BaseIcon )`
+	background-color: ${ ( props ) => props.theme.colors.surface };
+`;
+
+export default function CheckoutNextSteps( { responseCart, headerText }: Props ): JSX.Element {
+	const translate = useTranslate();
+
+	const steps: NextStepsStep[] = useMemo( () => {
+		if ( hasDIFMProduct( responseCart ) ) {
+			return [
+				{
+					text: translate( 'Submit business information' ),
+					icon: <CompletedStepIcon />,
+				},
+				{
+					text: translate( 'Choose a design' ),
+					icon: <CompletedStepIcon />,
+				},
+				{
+					text: <b>{ translate( 'Checkout' ) }</b>,
+					icon: <CurrentStepIcon />,
+				},
+				{
+					text: translate( 'Submit content for new site' ),
+					icon: <NextStepIcon />,
+				},
+				{
+					text: translate( 'Receive your finished site in under 4 business days!' ),
+					icon: <NextStepIcon />,
+				},
+			];
+		}
+		return [];
+	}, [ responseCart, translate ] );
+
+	return steps && steps.length ? (
+		<CheckoutNextStepsWrapper>
+			<CheckoutNextStepsHeader>
+				{ headerText || translate( "What's Next" ) }
+			</CheckoutNextStepsHeader>
+			<CheckoutNextStepsListWrapper>
+				{ steps.map( ( step ) => (
+					<CheckoutNextStepsListItem>
```

Comment:
Minor: I think this will need a `key` prop so that React can differentiate the items. Something like,

```suggestion
					<CheckoutNextStepsListItem key={ step.text } >
```

Although... weirdly I don't see any warnings about this in the browser while testing. I wonder if it's not required somehow? It still seems like it might be a good idea.

Reply:
Good catch! I missed this in my testing. I just tested it, and I did see the errors in my browser console. `eslint-react-plugin` has a [rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-key.md) for this, I wonder why it did not apply here. I'll look into this!